File: 000-assignments/01-intermediate-dsa/030-linked-list-basics/class-rectangle/rectangle.py
class Rectangle:
    # Define properties here
    width = 0
    height = 0
    
    # Define constructor here
    def __init__(self, width, height):
        self.width = width
        self.height = height

    def perimeter(self):
        return 2 * (self.width + self.height)
    
    def area(self):
        return self.width * self.height
    

File: 000-assignments/04-LLD/scaler-lld/src/main/java/_1_intro_to_oops/Rectangle.java
package _1_intro_to_oops;

public class Rectangle {
    Point topLeft;
    int height;
    int width;

    public Rectangle(Point topLeft, int height, int width) {
        this.topLeft = topLeft;
        this.height = height;
        this.width = width;
    }

    public int area() {
        return height * width;
    }

    public int perimeter() {
        return 2 * (height + width);
    }

    public Point getBottomRight() {
        Point bottomRight = new Point(1, 2);

        bottomRight.x = topLeft.x + width;
        bottomRight.y = topLeft.y + height;

        return bottomRight;
    }

    public static void main(String[] args) {

        Rectangle rectangle = new Rectangle(new Point(0, 0), 10, 20);
        System.out.println(rectangle.area());
        System.out.println(rectangle.perimeter());
    }
}
